portfolio/src/main/webapp/script.js

```diff
@@ -16,14 +16,58 @@ function getComments() {
   fetch('/data').then(response => response.json()).then((commentsArray) => {
     let container = document.getElementById('comments-container');
     container.innerText = '';
+    console.log(commentsArray);
     for (const comment of commentsArray) {
-        container.appendChild(createListElement(comment));
+        container.append(createCommentView(comment));
     }
   });
 }
 
+function initMap() {
+  const prefPlace = { lat: 55.17477877, lng: 61.29954625 };
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: prefPlace,
+    zoom: 12,
+    mapTypeId: 'hybrid'
+  });
+  marker = new google.maps.Marker({position: prefPlace, map: map});
+}
+
+function loadAutorizationView() {
+  fetch('/autorization').then(response => response.text()).then(data => {
```

Comment:
Did you consider using `async` and `await`? The continuation passing style is a bit more obtuse.

```javascript
  function loadAuthorizationView() {
    fetch('/authorization').then(response => response.text()).then(data => {
      ...
    })
  }
```

can become:

```javascript
  async function loadAuthorizationView() {
    const response = await fetch('/authorization');
    const data = await response.text();
    ...
  }
```

The second version doesn't require lots of nested functions. This comment also applies to processCommentViewStatus which can also do this.

Reply:
It doesn't really make much difference in this case, but there are reasons to prefer `async`/`await`.

If you have many steps, it is less nested:

```javascript
  function manyStepsContinuationPassingStyle() {
    // Note: returning the promise means you can compose the functions.
    return fetch('/foo')
        .then(response => response.text())
        .then(text => someAsyncWork(text))
        .then(result => someMoreAsyncWork(text));
  }
  async function manyStepsDirectStyle() {
    const response = await fetch('/foo');
    const text = await someAsyncWork(text);
    await someMoreAsyncWork(text);
  }
```

Also, it ensures that your functions are composable:

```javascript
  function nonComposableContinuationPassingStyle() {
    // to make this composable, you have to return the promise.
    fetch('/demo').then(response => response.text()).then(text => doSomethingWith(text));
  }
  async composableDirectStyle() {
    const response = await fetch('/demo');
    doSomethingWith(await text);
  }
```

Suppose that I wanted to write another function which called this one and then did something *after* we had done `doSomethingWith(text)`. For that to work, you need to either return the promise manually in the CPS version and use `myComposableContinuationPassingStyle(...).then(x => doSomethingAfter(x))`, or you can use `async`/`await` and `doSomethingAfter(await composableDirectStyle(...))`.